### ISSUE TITLE
SIGSEGV in boids example

### ISSUE THREAD

**Issue by nitingupta910** [gfx-rs/wgpu#3859]
**Description**

Running the boids example leads to SIGSEGV

**Repro steps**
```
cd wgpu
cargo build
gdb ./target/debug/boids
```

**Expected vs observed behavior**
```
Thread 1 "boids" received signal SIGSEGV, Segmentation fault.
```

**Extra materials**

Here's backtrace from gdb:
```
Thread 1 "boids" received signal SIGSEGV, Segmentation fault.
0x00007fffecbd94db in ?? () from /usr/lib/x86_64-linux-gnu/libvulkan_intel.so
(gdb) bt
#0  0x00007fffecbd94db in ?? () from /usr/lib/x86_64-linux-gnu/libvulkan_intel.so
#1  0x00007fffebadaa90 in vulkan_layer_chassis::CmdBeginDebugUtilsLabelEXT (commandBuffer=0x555557b98720, pLabelInfo=0x7fffffff8630)
    at /vulkan-sdk/1.3.243.0/source/Vulkan-ValidationLayers/layers/generated/chassis.cpp:10088
#2  0x000055555642672e in ash::extensions::ext::debug_utils::DebugUtils::cmd_begin_debug_utils_label (self=0x555556be86b8, command_buffer=..., label=0x7fffffff8630)
    at /home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/ash-0.37.3+1.3.251/src/extensions/ext/debug_utils.rs:71
#3  0x00005555562bfa71 in wgpu_hal::vulkan::command::{impl#2}::begin_debug_marker (self=0x555557b5c7f8, group_label=...) at wgpu-hal/src/vulkan/command.rs:566
#4  0x0000555556179801 in wgpu_core::global::Global<wgpu_core::identity::IdentityManagerFactory>::command_encoder_push_debug_group<wgpu_core::identity::IdentityManagerFactory, wgpu_hal::vulkan::Api> (self=0x5555578d6450, encoder_id=..., label=...) at wgpu-core/src/command/mod.rs:406
#5  0x00005555560000d2 in wgpu::backend::direct::{impl#7}::command_encoder_push_debug_group (self=0x5555578d6450, encoder=0x7fffffff89c8, encoder_data=0x555557b2b170, 
    label=...) at wgpu/src/backend/direct.rs:2008
#6  0x0000555555f24bf2 in wgpu::context::{impl#5}::command_encoder_push_debug_group<wgpu::backend::direct::Context> (self=0x5555578d6450, encoder=0x7fffffff8c80, 
    encoder_data=..., label=...) at wgpu/src/context.rs:2726
#7  0x0000555556007c42 in wgpu::CommandEncoder::push_debug_group (self=0x7fffffff8c68, label=...) at wgpu/src/lib.rs:2960
#8  0x00005555556e3a70 in boids::{impl#0}::render (self=0x7fffffffadb0, view=0x7fffffff96d0, device=0x7fffffffaeb0, queue=0x7fffffffaed8, _spawner=0x7ffffffface0)
    at examples/boids/src/main.rs:303
#9  0x0000555555717269 in wgpu_example::framework::start::{closure#0}<boids::Example> (event=..., control_flow=0x7fffffff9c70) at examples/common/src/framework.rs:376
#10 0x00005555556d8331 in winit::platform_impl::platform::sticky_exit_callback<(), wgpu_example::framework::start::{closure_env#0}<boids::Example>> (evt=..., 
    target=0x7fffffffb798, control_flow=0x7fffffff9c70, callback=0x7fffffffaca8)
    at /home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/mod.rs:849
#11 0x00005555556f9f02 in winit::platform_impl::platform::wayland::event_loop::EventLoop<()>::run_return<(), wgpu_example::framework::start::{closure_env#0}<boids::Example>> (self=0x7fffffffb748, callback=...)
    at /home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/wayland/event_loop/mod.rs:529
#12 0x00005555556faf51 in winit::platform_impl::platform::wayland::event_loop::EventLoop<()>::run<(), wgpu_example::framework::start::{closure_env#0}<boids::Example>>
    (self=..., callback=...) at /home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/wayland/event_loop/mod.rs:222
#13 0x00005555556d812c in winit::platform_impl::platform::EventLoop<()>::run<(), wgpu_example::framework::start::{closure_env#0}<boids::Example>> (self=..., 
    callback=...) at /home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/mod.rs:755
#14 0x00005555556aff1c in winit::event_loop::EventLoop<()>::run<(), wgpu_example::framework::start::{closure_env#0}<boids::Example>> (self=..., event_handler=...)
    at /home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/event_loop.rs:278
#15 0x0000555555716ace in wgpu_example::framework::start<boids::Example> () at examples/common/src/framework.rs:288
#16 0x000055555571495c in wgpu_example::framework::run<boids::Example> (title=...) at examples/common/src/framework.rs:441
#17 0x00005555556e3d82 in boids::main () at examples/boids/src/main.rs:326
```

**Platform**

wgpu trunk: commit 80b178fa56949bc5ab3d077fd4586e220ddc780c

Ubuntu 22.04.2 LTS
Intel(R) Core(TM) i5-9400H CPU @ 2.50GHz

GPU:
```
~/src/wgpu$ sudo lshw -c video
[sudo] password for ngupta: 
  *-display                 
       description: VGA compatible controller
       product: CoffeeLake-H GT2 [UHD Graphics 630]
       vendor: Intel Corporation
       physical id: 2
       bus info: pci@0000:00:02.0
       logical name: /dev/fb0
       version: 02
       width: 64 bits
       clock: 33MHz
       capabilities: pciexpress msi pm vga_controller bus_master cap_list rom fb
       configuration: depth=32 driver=i915 latency=0 resolution=3840,2160
       resources: irq:169 memory:cb000000-cbffffff memory:80000000-8fffffff ioport:3000(size=64) memory:c0000-dffff
```

**Comment by teoxoy** [comment]
This looks similar to https://github.com/gfx-rs/wgpu/issues/3377. Are you running the same mesa version?

**Comment by jimblandy** [comment]
If they're running Ubuntu 22.04.2 LTS that's almost certainly a broken version Mesa. The fix isn't even on the Mesa 23.0 branch.

Closing as duplicate of #3377.
